lib/controllers/share.js

```diff
@@ -67,6 +67,17 @@ function resolveGist(serviceOptions, id) {
         }
     });
 }
+/*
+  Generate short ID by hashing body, converting to base62 then truncating.
+ */
+function shortId(body, length) {
+    console.time('shortid');
+    var base62 = require("base-x")('0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ');
+    var buf = Buffer.from(body, 'utf-8');
+    var b = base62.encode(require('sha1')(buf));
+    console.timeEnd('shortid');
+    return b.slice(0, length || 16);
```

Comment:
If you just use all 28 (if I did the math right), it's not really all that different from 16 is it?  And then we don't need to decide the effect of that truncation on the possibility of collisions.  I'm ok with having the option to make it shorter, but let's just use the whole thing as the default.